agency/test/hooks/issuer-profile.test.ts

```diff
@@ -0,0 +1,53 @@
+import { HookContext } from '@feathersjs/feathers';
+import { checkValidIssuerProfile } from '../../src/utils/hooks/issuer-profile';
+import memory from 'feathers-memory';
+import app from '../../src/app';
+import { DuplicatedProfileError } from '../../src/models/errors';
+import { GeneralError } from '@feathersjs/errors';
+import profile from '../data/profile.json';
+
+describe('\'issuer-profile\' hooks', () => {
+  beforeEach(() => {
+    const options = {
+      paginate: app.get('paginate')
+    };
+    app.use('/issuer/model', memory(options));
+  });
+
+  it('should return a valid issuer profile', async () => {
+    await app.service('issuer/model').create(profile);
+
+    const context = {
+      params: {
+        headers: { 'issuer-api-key': 'valid-api-key' }
+      },
+      app
+    } as unknown as HookContext;
+    expect(await checkValidIssuerProfile(context)).toEqual(context);
+  });
+
+  it('should throw an error if a duplicate profile exists ', async () => {
+    await app.service('issuer/model').create(profile);
+    await app.service('issuer/model').create(profile);
```

Comment:
Shouldn't this be only called once? The test will actually happen at line `39` when you `expect` the new check to fail, as long as a profile with the same data already exists.

Reply:
This test ensures that the `DuplicateProfileError`, which is thrown by the `checkValidIssuerProfile` hook if it `find`s more than one of the same profiles. The only way to trigger this is to add the profile twice manually before checking the assertion.